lib/msal-common/src/telemetry/performance/PerformanceClient.ts

```diff
@@ -271,12 +271,14 @@ export abstract class PerformanceClient implements IPerformanceClient {
                         } else {
                             this.logger.verbose(`PerformanceClient: Submeasurement for ${measureName} already exists for ${current.name}, ignoring`, correlationId);
                         }
-                        if (current.accessTokenSize) {
-                            previous.accessTokenSize = current.accessTokenSize;
-                        }
-                        if (current.idTokenSize) {
-                            previous.idTokenSize = current.idTokenSize;
-                        }
+
+                        // Attach any additional submeasurement fields to the event to be emitted
+                        Object.keys(current).forEach(key => {
```

Comment:
While this would make things easier, it could make all events much larger in size, theoritcally.

Reply:
Also, when done this way, it is not possible to determine which properties came from which measurements (which is why the durationMs prefixes with the name of the submeasurement).

Reply:
I'd rather see an allow list of properties, and when needed, have them prefixed accordingly.

Reply:
> While this would make things easier, it could make all events much larger in size, theoritcally.

Intuitively speaking I think it makes sense for any fields included on an event to be emitted. If you don't intend for a field to be emitted on the event you shouldn't include it on the submeasurement.

> Also, when done this way, it is not possible to determine which properties came from which measurements (which is why the durationMs prefixes with the name of the submeasurement).

Prefixing makes sense for perf measurements but I'm not so sure the specific submeasurement is important for other data being passed through (and if some sort of distinction is necessary the fields themselves can be named appropriately like these `mats_` fields were)

Reply:
> Prefixing makes sense for perf measurements but I'm not so sure the specific submeasurement is important for other data being passed through (and if some sort of distinction is necessary the fields themselves can be named appropriately like these mats_ fields were)

One example is the `success` property, which for a given there may be multiple submeasurements that have `success: false`, but the top-level measurement is `success: true` (e.g. `acquireTokenSilent` where the cached RT has expired but we are able successfully acquire a new one).

Reply:
The point is that it should be clear for each property which specific operation the value was capture for, and that will not always be the case if all properties are capture without prefixing (or explicit allow-listing for certain well-known properties.

Reply:
So keep the loop but check if each property is in an allow list and if it's not prefix it with the submeasurement name. Does that work?

Reply:
I'd rather not add all events by default (even prefixed).

Reply:
> Intuitively speaking I think it makes sense for any fields included on an event to be emitted. If you don't intend for a field to be emitted on the event you shouldn't include it on the submeasurement.

This feels reasonable to me to add everything that isn't undefined.